e2e-test/server/serve_test.go

```diff
@@ -0,0 +1,55 @@
+package server_test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/sco1237896/sco-backend/cmd/serve"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestServe(t *testing.T) {
+	cmd := serve.NewServeCmd()
+	cmd.SetArgs([]string{"--bind-address", "localhost:9090"})
+
+	go func() {
+		err := cmd.ExecuteContext(context.Background())
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, "http://localhost:9090/pipes", http.NoBody)
+	if err != nil {
+		t.Error(err)
+	}
+
+	assertEventually(t, func() bool {
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			t.Error(err)
+		}
+
+		defer resp.Body.Close()
+		return http.StatusOK == resp.StatusCode
+	})
+}
+
+// wrapper for assert.Eventually.
+func assertEventually(t *testing.T, condition func() bool) {
```

Comment:
I would encourage to switch to [gomega](https://github.com/onsi/gomega) at some point